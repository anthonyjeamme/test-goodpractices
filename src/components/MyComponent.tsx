import React from 'react'

const MyComponent = ({ data }) => {
	if (!data) return null

	return <div className="MyComponent">Ton nom : {data.name}</div>
}
export default MyComponent
