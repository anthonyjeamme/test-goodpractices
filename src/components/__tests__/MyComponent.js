import React from "react"
import renderer from "react-test-renderer"
import MyComponent from "../MyComponent.tsx"

describe("Header", () => {
  it("renders correctly A", () => {
    const tree = renderer
      .create(<MyComponent data={{ name: "tony" }} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders correctly B", () => {
    const tree = renderer.create(<MyComponent />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
