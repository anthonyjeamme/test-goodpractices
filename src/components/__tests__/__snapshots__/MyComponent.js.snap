// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="MyComponent"
>
  Ton nom : 
  tony
</div>
`;

exports[`Header renders correctly 2`] = `null`;

exports[`Header renders correctly A 1`] = `
<div
  className="MyComponent"
>
  Ton nom : 
  tony
</div>
`;

exports[`Header renders correctly B 1`] = `null`;
